packages/hardhat/test/poolsTest.ts

```diff
@@ -0,0 +1,324 @@
+import {SignerWithAddress} from '@nomiclabs/hardhat-ethers/signers'
+import chai from 'chai'
+import chaiAsPromised from "chai-as-promised"
+import {solidity} from "ethereum-waffle"
+import {utils} from 'ethers'
+import {ethers} from 'hardhat'
+import {ERC20Stub, ERC20Stub__factory, Pools, Pools__factory} from '../typechain'
+
+chai.use(solidity).use(chaiAsPromised)
+const {expect} = chai
+const {parseUnits, formatUnits} = utils
+
+describe("Pools", async () => {
+
+    let owner: SignerWithAddress
+    let lp1: SignerWithAddress
+    let lp2: SignerWithAddress
+    let uniswap: string
+    let poolsFactory: Pools__factory
+
+    let erc20: ERC20Stub
+
+    let sut: Pools
+
+    beforeEach(async () => {
+
+        let signers = await ethers.getSigners()
+        owner = signers[0]
+        lp1 = signers[1]
+        lp2 = signers[2]
+
+        uniswap = "0xE592427A0AEce92De3Edee1F18E0157C05861564"
+
+        const erc20Factory = (await ethers.getContractFactory('ERC20Stub', owner)) as ERC20Stub__factory
+        erc20 = await erc20Factory.deploy("Wrapped ETH", "WETH")
```

Comment:
I'd add at least 1 test (ideally more) that use more than one token, so you can prove the thing doesn't mix up balances and stuff